server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -523,26 +524,16 @@ protected void doRun() {
                         throw new IllegalArgumentException("only write ops with an op_type of create are allowed in data streams");
                     }
 
-                    IndexRouting indexRouting = concreteIndices.routing(concreteIndex);
-
-                    switch (docWriteRequest.opType()) {
-                        case CREATE:
-                        case INDEX:
-                            prohibitAppendWritesInBackingIndices(docWriteRequest, metadata);
-                            prohibitCustomRoutingOnDataStream(docWriteRequest, metadata);
-                            IndexRequest indexRequest = (IndexRequest) docWriteRequest;
-                            indexRequest.resolveRouting(metadata);
```

Comment:
I think the `resolveRouting(...)` method can now also be removed from `IndexRequest` class? (looks this was the only usage).

Reply:
Doh! I had done that when I was working on it and had lost it in the shuffle. I'll add it back